src/chainstate/stacks/boot/pox.clar

```diff
@@ -153,8 +169,33 @@
             )
         ;; no state at all
         none
-    )
-)
+    ))
+
+(define-private (check-caller-allowed)
+    (or (is-eq tx-sender contract-caller)
+        (let ((caller-allowed 
+                 ;; if not in the caller map, return false
+                 (unwrap! (map-get? allowance-contract-callers
+                                    { sender: tx-sender, contract-caller: contract-caller })
+                          false)))
+          ;; is the caller allowance expired?
+          (if (< burn-block-height (unwrap! (get until-burn-ht caller-allowed) true))
+              (begin (map-delete allowance-contract-callers
+                                 { sender: tx-sender, contract-caller: contract-caller })
+                     false)
+              true))))
+
+(define-private (get-check-delegation (stacker principal))
+    (let ((delegation-info (try! (map-get? delegation-state { stacker: stacker }))))
+      ;; did the existing delegation expire?
+      (if (match (get until-burn-ht delegation-info)
+                 until-burn-ht (> burn-block-height until-burn-ht)
+                 false)
+          ;; it expired, delete the entry and return none
+          (begin (map-delete delegation-state { stacker: stacker })
```

Comment:
Same question here -- why should we ever delete?  Also, it seems wrong to me to be mutating state on a "getter" method.

Reply:
Same as above -- these are expired entries, and it keeps the contract from having to continuously re-check expired entries. Ultimately, there's not really a performance benefit to removing these, because the MARF lookup is so much more expensive than parsing the returned tuple and checking the block height. If you'd prefer, we can just leave expired entries in the map.

Reply:
I think if we can avoid doing work, we should -- it makes the contract cheaper without sacrificing correctness.

Also, avoiding mutations in both this and the other getter will make it possible to invoke these methods in a read-only manner.

Reply:
Okay, removed in 49605a7